client/js/contextMenuFactory.js

```diff
@@ -0,0 +1,193 @@
+"use strict";
+const $ = require("jquery");
+const socket = require("./socket");
+const utils = require("./utils");
+const JoinChannel = require("./join-channel");
+const ContextMenu = require("./contextMenu");
+const contextMenuActions = [];
+const contextMenuItems = [];
+
+module.exports = {
+	addContextMenuItem,
+	createContextMenu,
+};
+
+addDefaultItems();
+
+/**
+ *
+ * @param opts
+ * @param {function(Object)} [opts.check] - Function to check whether item should show on the context menu, called with the target jquery element, shows if return is truthy
+ * @param {string|function(Object)} opts.className - class name for the menu item, should be prefixed for non-default menu items (if function, called with jquery element, and uses return value)
+ * @param {string|function(Object)} opts.data - data that will be sent to the callback function (if function, called with jquery element, and uses return value)
+ * @param {string|function(Object)} opts.display - text to display on the menu item (if function, called with jquery element, and uses return value)
+ * @param {boolean} [opts.divider] - Whether to put a divider after this option in the menu
+ * @param {function(Object)} opts.callback - Function to call when the context menu item is clicked, called with the data requested in opts.data
+ */
+function addContextMenuItem(opts) {
+	opts.check = opts.check || (() => true);
+	opts.actionId = contextMenuActions.push(opts.callback) - 1;
+	contextMenuItems.push(opts);
+}
+
+function createContextMenu(that, event) {
+	return new ContextMenu(contextMenuItems, contextMenuActions, that, event);
+}
+
+function addWhoisItem() {
+	function whois(itemData) {
+		const chan = utils.findCurrentNetworkChan(itemData);
+
+		if (chan.length) {
+			chan.click();
+		}
+
+		socket.emit("input", {
+			target: $("#chat").data("id"),
+			text: "/whois " + itemData,
+		});
+
+		$(`.channel.active .users .user[data-name="${itemData}"]`).click();
+	}
+
+	addContextMenuItem({
+		check: (target) => target.hasClass("user"),
+		className: "user",
+		display: (target) => target.text(),
```

Comment:
I didn't want to make a separate PR to reduce the rebase hell for you, but could you change this to `target.data("name")` to match `data`?

If you open context menu on a user where they sent a message as op, their nick will show up as `@xPaw` in the context menu.

Reply:
Done, and squashed.